lib/mods_display/fields/name.rb

```diff
@@ -163,23 +164,37 @@ def rebuild_fields_with_new_labels(label_keys, results)
       # Build the new fields data, stripping out the roles within the Person classes
       label_keys.uniq.map do |k|
         values = results[k].map do |person|
-          ModsDisplay::Name::Person.new(name: person.name)
+          ModsDisplay::Name::Person.new(name: person.name, orcid: person.orcid)
         end
 
         ModsDisplay::Values.new(label: k, values: values)
       end
     end
 
     class Person
-      attr_accessor :name
+      attr_accessor :name, :orcid
 
       def initialize(data)
         @name = data[:name]
+        @orcid = if data[:orcid].present?
+                   data[:orcid]
+                 elsif data[:name_identifiers].present?
+                   orcid_identifier(data[:name_identifiers])
+                 end
       end
 
       def to_s
         @name
       end
+
+      private
+
+      def orcid_identifier(name_identifiers)
+        orcid = name_identifiers.filter do |name_identifier|
```

Comment:
Can you use `.select` to be consistent with above code (given that `.filter` is just an alias)?

Reply:
yup - good idea.  i was toying with filter_map at one point, hence the filter in the first place. Done.